### ISSUE TITLE
Content `null` handling not working for root values

### ISSUE THREAD

**Issue by joca-bt** [FasterXML/jackson-databind#3227]
I am trying to build an ObjectMapper which doesn't allow nulls. I noticed some slight discrepancy in behaviour depending on whether I am deserializing to a custom class representing a JSON object or something else.

```
public class Input {
    public Boolean field;
}

ObjectMapper objectMapper = JsonMapper.builder()
    .defaultSetterInfo(JsonSetter.Value.construct(Nulls.FAIL, Nulls.FAIL))
    .build();

objectMapper.readValue("null", Boolean.class); // 1) returns null
objectMapper.readValue("null", boolean.class); // 2) returns null
objectMapper.readValue("{ \"field\": null }", Map.class); // 3) returns Map.of("field", null)
objectMapper.readValue("{ \"field\": null }", Input.class); // 4) throws error
```

I was trying to generate an error for all cases but only getting for 4). Is this expected?

**Comment by cowtowncoder** [comment]
Behavior for cases 1 and 2 is as expected because `JsonSetter` only refers to properties (main setting) or "content" (values of Map entries) (second setting).

But that does not explain your case 3. So that seems like a possible bug.

**Comment by cowtowncoder** [comment]
Thanks! Merged -- I have some concerns on this not working for per-type config overrides, but I guess we can cross that bridge when we get there.

**Comment by cowtowncoder** [comment]
Merged the patch, will be fixed in 2.13.0.

**Comment by joca-bt** [comment]
@cowtowncoder this has been fixed and can be closed (#3 and #4 both return error).

**Comment by cowtowncoder** [comment]
@joca-bt Thank you for pointing it out, yes, it's included in release notes but accidentally left open here.
